File: TP01/C/aleatorio.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>


  /*Método para checar se duas Strings são idênticas
   *@param: String s1, String s2
   *@return: Booleano T||F
   */
	bool strCmpr(char* str1, char* str2){
		bool resp = true;
		if(strlen(str1) != strlen(str2)){
			resp = false;														//Condição para evitar erro de indexação devido a tamanhos diferentes de char*
		}
		else{																	//Compara caracter por caracter até encontrar um diferente, de tal forma retorna falso se achado
			for(int i = 0; i < strlen(str1); i++){			
				if(str1[i] != str2[i]){
					resp = false;
					i = strlen(str1);											//Saída lógica do loop
				}
			}
		}
		return resp;
	}


	/*Método para alterar toda ocorrencia de um certo caracter da cadeia de chars (char*) com parametros aleatorios
	 *@param: char* s para ser alterada
	 *@return: char* s com as modificações
	 */


	char* gerador (char* s){

		int size = strlen(s);
		
		char* auxiliar = s;

   	 	char a, b;
    	a = 'a' + abs(rand()) % 26;
    	b = 'a' + abs(rand()) % 26;

		for(int i = 0; i < size; i++) {
        	if(s[i] == a) {
            	auxiliar[i] = b;
        	}
   		}

   		return auxiliar;	
	}




int main(){
		
	char* str = (char*) malloc(100*sizeof(char));

	srand(4);
	
	while(strcmp(str,"FIM") != 0){
		fgets(str,100,stdin);
		size_t len = strlen(str);
		if (len > 0 && str[len-1] == '\n') {			//lógica para garantir que a String é terminada da forma correta para sofrer a manipulação 
		    str[len-1] = '\0';
		}	
		if(strcmp(str,"FIM") != 0){
			printf("%s\n",gerador(str));
		}
	}
	return 0;
}

File: TP01/Java/Aleatorio.java
/*
@author: Gabriel Samarane Ribeiro
@Atividade do TP1 para AEDS2 2023/2
*/
import java.util.Random;
class Aleatorio{

	/*Método para alterar toda ocorrencia de um certo caracter da String com parametros aleatorios
	 *@param: String s para ser alterada
	 *@return: String s com as modificações
	 */

	public static String randomCharOnStr(String str){
		char[] array = str.toCharArray();
		Random gerador = new Random();
		gerador.setSeed(4);
		char a, b;
		a = (char)('a' + Math.abs(gerador.nextInt()) % 26);
		b = (char)('a' + Math.abs(gerador.nextInt()) % 26);
		for(int i = 0; i < str.length(); i++){
			if(str.charAt(i) == a){
			array[i] = b;
			}
		}
		return new String(array);
	
	}

	
  /*Método para checar se duas Strings são idênticas
    @param: String s1, String s2
    @return: Booleano T||F
  */
	public static boolean strCmpr(String str1, String str2){
		boolean resp = true;
		if(str1.length() != str2.length()){
			resp = false;	//condição para evitar erro de indexação devido a tamanhos diferentes de String
		}
		else{																	//Compara caracter por caracter até encontrar um diferente, de tal forma retorna falso se achado
			for(int i = 0; i < str1.length(); i++){					
				if(str1.charAt(i) != str2.charAt(i)){
					resp = false;
				}
			}
		}
		return resp;
	}


	//Método principal

	public static void main(String args[]){

		String str = "";

		while(!strCmpr(str,"FIM")){	
			str = MyIO.readLine();
			if(!strCmpr(str,"FIM")){
				MyIO.print(randomCharOnStr(str));
				MyIO.print("\n");	
			
			}	
		
		}

	}
}
